watchtower/lookout/justice_descriptor_test.go

```diff
@@ -111,9 +119,54 @@ func testJusticeDescriptor(t *testing.T, blobType blob.Type) {
 	toLocalScriptHash, err := input.WitnessScriptHash(toLocalScript)
 	require.Nil(t, err)
 
-	// Compute the to-remote witness script hash.
-	toRemoteScriptHash, err := input.CommitScriptUnencumbered(toRemotePK)
-	require.Nil(t, err)
+	// Compute the to-remote redeem script, witness script hash, and
+	// sequence numbers.
+	//
+	// NOTE: This is pretty subtle.
+	//
+	// The actual redeem script for a p2wkh output is just the pubkey, but
+	// the witness sighash calculation injects the classic p2kh script:
+	// OP_DUP OP_HASH160 <pubkey-hash160> OP_EQUALVERIFY OP_CHECKSIG. When
```

Comment:
Heh yeh this can be super confusing for those that haven't read BIP 141 closely and/or implemented the _verification_ on the script level....

Reply:
the confusing part is the sign descriptor accepts a `WitnessScript`, which can be either a redeem script or a witness program.